naga/src/valid/analyzer.rs

```diff
@@ -15,19 +16,15 @@ use std::ops;
 
 pub type NonUniformResult = Option<Handle<crate::Expression>>;
 
-// Remove this once we update our uniformity analysis and
-// add support for the `derivative_uniformity` diagnostic
-const DISABLE_UNIFORMITY_REQ_FOR_FRAGMENT_STAGE: bool = true;
-
```

Comment:
Hmm. This kludge was serving two purposes:
1) Cover for our inability to parse the directives (fixed by this PR).
2) Cover for the over-strictness of our analysis (not fixed by this PR).

I understand that we need to remove this if we want to be able to test the application of the directives, but I hope that point 2) doesn't cause us trouble.

Reply:
I think we're in the clear with (2). If WGPU users run into errors that are disjoint with the spec., they should be able to `diagnostic(off, derivative_uniformity);`. This should work even in fully compliant environments, so I think this will be annoying for end users to deal with, but it will ultimately let them ship in more places.

Reply:
Fixed with the force-pushed 1ffd9c26366cfe7aabc06bc29884bb52740d1076.